applications/MappingApplication/custom_mappers/coupling_geometry_mapper.cpp

```diff
@@ -285,61 +286,58 @@ void CouplingGeometryMapper<TSparseSpace, TDenseSpace>::EnforceConsistencyWithSc
+    {
+        if (std::abs(slave_row_sums_vector[row_counter]/ projected_row_sums_vector[row_counter] - 1.0) > 1e-15)
+        {
+            // Correct entries
+            const double alpha = (slave_row_sums_vector[row_counter] / projected_row_sums_vector[row_counter] < scalingLimit)
+                ? slave_row_sums_vector[row_counter] / projected_row_sums_vector[row_counter] : scalingLimit;
+            for (auto col_it = row_it.begin(); col_it != row_it.end(); ++col_it) (*col_it) *= alpha;
+        }
+        ++row_counter;
+    }
 }
 
 template<class TSparseSpace, class TDenseSpace>
 void CouplingGeometryMapper<TSparseSpace, TDenseSpace>::CalculateMappingMatrixWithSolver(
     MappingMatrixType& rConsistentInterfaceMatrix, MappingMatrixType& rProjectedInterfaceMatrix)
 {
-    // mpMappingMatrix = Kratos::make_unique<DenseMappingMatrixType>(DenseMappingMatrixType(
-    //     rConsistentInterfaceMatrix.size1(), rProjectedInterfaceMatrix.size2()));
-
-    // const size_t n_rows = mpMappingMatrix->size1();
-    // #pragma omp parallel
-    // {
-    //     Vector solution(n_rows);
-    //     Vector projector_column(n_rows);
-
-    //     #pragma omp for
-    //     for (int i = 0; i < static_cast<int>(mpMappingMatrix->size2()); ++i)
-    //     {
-    //         for (size_t j = 0; j < n_rows; ++j) projector_column[j] = rProjectedInterfaceMatrix(j, i);
-    //         mpLinearSolver->Solve(rConsistentInterfaceMatrix, solution, projector_column);
-    //         for (size_t j = 0; j < n_rows; ++j) (*mpMappingMatrix)(j, i) = solution[j];
-    //     }
-    // }
-}
+    mpMappingMatrix = Kratos::make_unique<typename SparseSpaceType::MatrixType>(
+        rConsistentInterfaceMatrix.size1(),
+        rProjectedInterfaceMatrix.size2());
 
-template<class TSparseSpace, class TDenseSpace>
-void CouplingGeometryMapper<TSparseSpace, TDenseSpace>::CheckMappingMatrixConsistency()
-{
-    // @Peter this is already included in "MappingMatrixUtilities::CheckRowSum"
-
-    // for (size_t row = 0; row < mpMappingMatrix->size1(); ++row) {
-    //     double row_sum = 0.0;
-    //     for (size_t col = 0; col < mpMappingMatrix->size2(); ++col) row_sum += (*mpMappingMatrix)(row, col);
-    //     if (std::abs(row_sum - 1.0) > 1e-12) {
-    //         KRATOS_WATCH(*mpMappingMatrix)
-    //         KRATOS_WATCH(row_sum)
-    //         KRATOS_ERROR << "mapping matrix is not consistent\n";
-    //     }
-    // }
+    const size_t n_rows = mpMappingMatrix->size1();
+    #pragma omp parallel
+    {
+        Vector solution(n_rows);
+        Vector projector_column(n_rows);
+
+        #pragma omp for
+        for (int i = 0; i < static_cast<int>(mpMappingMatrix->size2()); ++i)
+        {
+            for (size_t j = 0; j < n_rows; ++j) projector_column[j] = rProjectedInterfaceMatrix(j, i); // TODO try boost slice or project
+            mpLinearSolver->Solve(rConsistentInterfaceMatrix, solution, projector_column);
```

Comment:
@peterjwilson I am not sure if it is this is threadsafe to do in parallel, I am pretty sure it is not, depending on the linear solver used